File: PocketEdition/RoleRollsPocketEdition/Rolls/Application/RollInput.cs
using RoleRollsPocketEdition.Rolls.Domain.Entities;

namespace RoleRollsPocketEdition.Rolls.Application
{
    public class RollInput
    {
        public Guid PropertyId { get; set; }
        public RollPropertyType PropertyType { get; set; }

        public int Complexity { get; set; }
        public int Difficulty { get; set; }
        public int PropertyBonus { get; set; }
        public int RollBonus { get; set; }
        public bool Hidden { get; set; }

        public string Description { get; set; }
        public List<int> Rolls { get; set; } = new List<int>();

    }
}

File: web/src/app/pocket-role-rolls/campaigns/models/RollInput.ts
import {RollOrigin} from "../CampaignInstance/campaign-heroes/RollOrigin";

export class RollInput {
  public propertyName: string;
  public propertyValue: number;
  public propertyId: string;
  public propertyType: RollOrigin;
  public complexity: number;
  public difficulty: number;
  public propertyBonus: number;
  public rollBonus: number;
  public hidden: boolean;
  public creatureId: string;
  public rollsAsString = '';
  public rolls: number[];
  public description: string;
}
